dashboard/src/utils/filterDataset.js

```diff
@@ -0,0 +1,10 @@
+export const filterData = (dataArray, startDate, endDate, controllerValue) => {
+  let res = dataArray.filter((data) => {
+    return (
+      data.controller.includes(controllerValue) &&
+      new Date(data.metadata["dataset.created"].split(":")[0]) >= startDate &&
+      new Date(data.metadata["dataset.created"].split(":")[0]) <= new Date(endDate)
+    );
+  });
```

Comment:
I think you can dispense with the braces and the `return` statement:
```
  let res = dataArray.filter(data =>
    data.controller.includes(controllerValue) &&
    new Date(data.metadata["dataset.created"].split(":")[0]) >= startDate &&
    new Date(data.metadata["dataset.created"].split(":")[0]) <= new Date(endDate)
  );
```

Reply:
Although, it might be worth capturing the value of `new Date(data.metadata["dataset.created"].split(":")[0])` rather than producing it a second time, and for that you'll probably need the braces and the `return` statement. 😕